rlbot_gui/gui/js/main-vue.js

```diff
@@ -21,13 +21,30 @@ export default {
 
 		<b-navbar-nav class="ml-auto">
 			<b-spinner v-if="showProgressSpinner" variant="success" label="Spinning" class="mr-2"></b-spinner>
-			<span v-if="!matchSettings.enable_state_setting" class="mr-2">
+			<span id="sandbox-button-wrapper">
+				<b-button
+					@click="$router.replace('/sandbox')" variant="dark"
+					:disabled="!matchSettings.enable_state_setting">
+					State Setting Sandbox
+				</b-button>
+			</span>
+			<b-tooltip target="sandbox-button-wrapper" v-if="!matchSettings.enable_state_setting">
 				<b-icon class="warning-icon" icon="exclamation-triangle-fill"></b-icon>
 				State setting is turned off, sandbox won't work!
+			</b-tooltip>
+
+			<span id="story-button-wrapper">
+				<b-button 
+					@click="$router.replace('/story')" variant="dark" class="ml-2"
+					:disabled="!hasAValidFolder()">
+					Story Mode
+				</b-button>
 			</span>
-			<b-button @click="$router.replace('/sandbox')" variant="dark">
-				State Setting Sandbox
-			</b-button>
+			<b-tooltip target="story-button-wrapper" v-if="!hasAValidFolder()">
+				<b-icon class="warning-icon" icon="exclamation-triangle-fill"></b-icon>
+				Download the BotPack first!
+			</b-tooltip>
```

Comment:
Good solution in terms of user interface! You should probably make a more specific function than hasAValidFolder though. That will turn true if somebody manually loads a folder, uses the "create new bot" feature, etc.

We have some code that checks whether a botpack update is available, maybe that could inspire a botpack existence check.

Reply:
I switched to doing that. It uses the COMMIT_ID. Nothing unsets it though so there will be edge cases but it should be good enough in most cases

Reply:
Sounds good